pkg/instance/service.go

```diff
@@ -661,6 +664,79 @@ func (s service) FindInstances(user *model.User, presets bool) ([]GroupsWithInst
 	return instances, err
 }
 
+type InstanceStatus string
+
+const (
+	NotDeployed        InstanceStatus = "NotDeployed"
+	Pending            InstanceStatus = "Pending"
+	Booting            InstanceStatus = "Booting"
+	BootingWithRestart InstanceStatus = "Booting (%d)"
+	Running            InstanceStatus = "Running"
+	Error              InstanceStatus = "Error"
+)
+
+func (s service) GetStatus(instance *model.Instance) (InstanceStatus, error) {
+	ks, err := NewKubernetesService(instance.Group.ClusterConfiguration)
+	if err != nil {
+		return "", err
+	}
+
+	pod, err := ks.getPod(instance, "")
+	if err != nil {
+		if strings.HasPrefix(err.Error(), "no pod found using the selector: ") {
+			return NotDeployed, nil
+		}
+		return "", err
```

Comment:
Wrap this error? `fmt.Errorf("failed to get pod: %v", err)`

Reply:
The error is already wrapped in the `getPod` method. The only reason for having the if statement here is because we don't want to return an error if the pod isn't found. It could simply be that it's not deployed.

I was wondering if it would be better to return errdef.NotFound rather than checking the error message though.

Reply:
> I was wondering if it would be better to return errdef.NotFound rather than checking the error message though.

I think that makes sense, yeah.